### PR TITLE
Replace ultra runner by turborepo

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Replacing ultra-runner by turborepo.

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by gdostie** [comment]
Turbo cache on github runners works for subsequent runs 🎉 
image
https://github.com/coveo/plasma/actions/runs/3252780928/jobs/5339365742#step:4:43

**Comment by gdostie** [comment]
Turbo cache activated for build and test commands.

Cache miss (happens on initial PR build)

image

Cache hit (happens on PR updates)

image

File: .github/actions/test/action.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ runs:
     using: composite
     steps:
         - shell: bash
-          run: pnpm --workspace-concurrency 1 test:ci
+          run: pnpm test
```
Comment (aiheon) [inline .github/actions/test/action.yml:7]: `test:ci` was removing all the logs that we don't really care about, are you sure you want to use `test` instead? 🤔

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
         "reconstruct": "npx rimraf node_modules packages/*/node_modules && npm run setup && echo done",
         "setup": "pnpm install",
         "start": "pnpm build && pnpm --recursive --parallel start",
-        "test": "pnpm --recursive test",
+        "test": "turbo run test",
```
Comment (aiheon) [inline package.json:16]: Now I want this in the admin 👀

**Review by aiheon — APPROVED** [review]

File: .github/actions/test/action.yml (hunk 1 above)
Comment (gdostie) [reply .github/actions/test/action.yml:7]: `test` leverages turbo,  `test:ci` does not so yes I am sure. What I'm unsure of is how to pass down arguments through turborepo commands

File: package.json (hunk 2 above)
Comment (gdostie) [reply package.json:16]: turbo is already in the admin, it would be very easy to add it for the test command!

Comment (aiheon) [reply package.json:16]: I think I will 😎

File: .github/actions/test/action.yml (hunk 1 above)
Comment (gdostie) [reply .github/actions/test/action.yml]: found it!, removed test:ci and test:changed

**Pull request merged**